torch/distributed/tensor/parallel/style.py

```diff
@@ -325,7 +367,140 @@ def make_output_reshard_tensor(
         A :class:`torch.Tensor` object converted from output DTensor.
     """
 
-    return make_output_shard_1d(output, device_mesh).to_local()  # type: ignore[call-arg, attr-defined]
+    return make_output_shard_1d(output, device_mesh).to_local()  # type: ignore[call-arg, attr-defined, misc]
+
+
+def _get_prepare_input(
```

Comment:
question about these functions: are we installing hooks by still using the prepare_input/prepare_output functions, that's why we need to generate prepare_input_output functions?

Reply:
that's the contract from the `distribute_module` API.

https://github.com/pytorch/pytorch/blob/8f90be447800a5c624d96b928387cb6f21509d37/torch/distributed/_tensor/api.py#L595

Reply:
I know, what I mean is that you are reusing the `_prepare_input_fn` to pass to distribute_module instead of generate a new function?

Reply:
right.